core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonSecuredHttpContext.java

```diff
@@ -178,16 +181,22 @@ public boolean handleSecurity(HttpServletRequest request, HttpServletResponse re
             return true;
         }
 
-        String indexPageURL = CarbonUIUtil.getIndexPageURL(session.getServletContext(),
-                request.getSession());
+        String indexPageURL;
+        if (!isConfigurableIndexPageURLFeatureEnabled()) {
+            indexPageURL = getDefaultIndexPageURL(session.getServletContext());
+        } else {
+            indexPageURL = CarbonUIUtil.getIndexPageURL(session.getServletContext(),
+                    request.getSession());
+
+            // Reading the requestedURL from the cookie to obtain the request made while not
+            // authanticated; and setting it as the indexPageURL
+            indexPageURL = CarbonUILoginUtil.getIndexPageUrlFromCookie(requestedURI, indexPageURL,
+                    request);
 
-        // Reading the requestedURL from the cookie to obtain the request made while not
-        // authanticated; and setting it as the indexPageURL
-        indexPageURL = CarbonUILoginUtil.getIndexPageUrlFromCookie(requestedURI, indexPageURL,
```

Comment:
We need to preserve this capability..

Having custom index page is a one capability.
Ability to return, last accessed page is another capability. Seems like this code is handling this case as well.. Even we can remove the above functionality(having custom index page) based on the flag, we cannot remove this one.

Reply:
Addressed in f42bc30b6acf6c38add848cc5253f24402ac6e90, by changing the approach.